include/boost/math/statistics/linear_regression.hpp

```diff
@@ -46,35 +49,29 @@ auto simple_ordinary_least_squares(RandomAccessContainer const & x,
     return std::make_pair(c0, c1);
 }
 
-template<class RandomAccessContainer>
-auto simple_ordinary_least_squares_with_R_squared(RandomAccessContainer const & x,
-                                   RandomAccessContainer const & y)
+template<typename RandomAccessContainer, typename Real = typename RandomAccessContainer::value_type, 
+         typename Size = typename RandomAccessContainer::size_type>
+auto simple_ordinary_least_squares_with_R_squared(RandomAccessContainer const & x, RandomAccessContainer const & y)
+    -> std::tuple<Real, Real, Real>
 {
-    using Real = typename RandomAccessContainer::value_type;
-    if (x.size() <= 1)
-    {
-        throw std::domain_error("At least 2 samples are required to perform a linear regression.");
-    }
-
-    if (x.size() != y.size())
-    {
-        throw std::domain_error("The same number of samples must be in the independent and dependent variable.");
-    }
-    auto [mu_x, mu_y, cov_xy] = boost::math::statistics::means_and_covariance(x, y);
-
-    auto var_x = boost::math::statistics::variance(x);
+    BOOST_ASSERT_MSG(std::size(x) > 2, "At least 2 samples are required to perform a linear regression.");
```

Comment:
I'm not sure that asserts are correct here: asserts normally verify our own logic, where these are checking input values.  In particular these checks disappear in release mode which rather worries me here.

Reply:
Makes sense; I'll revert the asserts.

As an aside has anything gone in recently that would have touched [tr1](https://ci.appveyor.com/project/jzmaddock/math/builds/37002310/job/lhb7gba4fatmsyen?fullLog=true#L1302) or [nonfinite_num_facets](https://ci.appveyor.com/project/jzmaddock/math/builds/37002310/job/5e2eaaq6o4gjb8gc?fullLog=true#L3885)? CI is showing a litany of failures with those two (see links). tr1 is ODR linker errors.

Reply:
Nothing has changed at our end - the serialization errors look like they're coming from that lib, the link errors are more worrying as they suggest that lib and exe were built with differing build options.

Reply:
I can reproduce after updating Boost.Build to latest develop, but have no idea what's going on yet :(

Reply:
Right before this went to CI Boost.Build had [two](https://github.com/boostorg/build/commit/30a7e738ad2757b811720a56173dd3ba3453eaef) [commits](https://github.com/boostorg/build/commit/8d37b68e1db40ef05ee974e6f263071cd42e6d79) on develop that changed linking on MSVC.